sqlserver/datadog_checks/sqlserver/connection.py

```diff
@@ -294,6 +294,34 @@ def _conn_key(self, db_key, db_name=None, key_prefix=None):
             key_prefix = ""
         return '{}{}:{}:{}:{}:{}:{}'.format(key_prefix, dsn, host, username, password, database, driver)
 
+    def _parse_connection_string(self, cs):
+        """
+        Parses a connection string (i.e. "key1=value1;key2=value2;...") into a map of {key -> value}
+        The keys are lower-cased as they're not case sensitive.
+        """
+        result = {}
+        for option in cs.split(';'):
+            option = option.strip()
+            if not option:
+                # strings can end with a semicolon in which case we'll have one empty option
+                continue
+            splits = [s.strip() for s in option.split('=')]
+            if len(splits) == 1:
+                raise ConfigurationError(
+                    "invalid connection_string as one of the options does not contain a '=': {}".format(option)
+                )
+            if len(splits) > 2:
+                raise ConfigurationError(
+                    "invalid connection_string as one of the options contains more than one '=': {}".format(option)
+                )
+            key, val = splits
+            if not key or not val:
+                raise ConfigurationError(
+                    "invalid connection_string option as it has an empty key or value: {}".format(option)
+                )
+            result[key.lower()] = val
```

Comment:
I think it's better to preserve the raw values as-is in the structure and do case insensitive comparison separately. For instance, when we want to display errors about the specific key that is incorrect, we have lost the original data at this point and customers will see a different value than they input.

Reply:
Yeah, updated to preserve for the initial parse, where we do now throw errors pointing to the exact index of the cs where we found the problem.